runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -156,4 +184,23 @@ private String uniqify(String baseName, Set<String> existing) {
   RunnerApi.Components toComponents() {
     return componentsBuilder.build();
   }
+
+  /**
+   * Checks to ensure that all elements that were registered are present within the components
+   * returned by {@link #toComponents()}.
+   */
+  public void validate() {
+    Map<AppliedPTransform<?, ?, ?>, String> missingTransforms = new HashMap<>();
+    for (Entry<AppliedPTransform<?, ?, ?>, String> registeredTransform : transformIds.entrySet()) {
+      if (!componentsBuilder.getTransformsMap().containsKey(registeredTransform.getValue())) {
+        missingTransforms.put(
+            registeredTransform.getKey(), registeredTransform.getValue());
+      }
+    }
+    checkState(
+        missingTransforms.isEmpty(),
+        "%s transforms were registered but were never added to components. Missing transform %s",
```

Comment:
Is there a way to get it so that registerPTransform(AppliedPTransform, List) is always invoked (with an empty list representing no children) instead of needing this validate method?